### PR TITLE
Simpler instructions wrt GitHub (and Nix for VM users)

### PR DESCRIPTION
**Pull request by lhstrh**
There simplifications are possible as we decided to:
 - add `gh` as a prerequisite because it's really convenient; and
 - include the dependencies normally managed by `nix` in the VM image.

### CODE REVIEW

File: src/GettingStarted.md [hunk 1]
```diff
@@ -38,31 +25,37 @@ Give your repo a name and click on "Create repository":
...
-Or, for a setup that uses GitHub CLI, run:
+On the command line on your host machine, change directory to the location where you would like to check out your repository. Let us assume that you named your repo `my-3pi`. Check it out using the following command (where `<username>` must be substituted with your GitHub username):
 
 ```bash
 $ gh repo clone <username>/<reponame>
 ```
 
+This will create a directory called `my-3pi` in the current working directory.
+
 The template includes [raspberrypi/pico-sdk](https://github.com/raspberrypi/pico-sdk) as a submodule, which itself also has a lot of submodules. We recommend against using the `--recursive` flag because we do not need to recursively clone the submodules inside of `pico-sdk`. Instead, change directory into the root of your clone and run:
```
Comment (victoria-tuck) [inline src/GettingStarted.md:36]: "Instead, change directory into the root of your clone and run the submodule update:"
The way it's written makes it seem like they should change directory and then cd again.

Comment (lhstrh) [reply src/GettingStarted.md:36]: Good catch! I just removed the extra `cd` from the code listing...

File: src/GettingStarted.md [hunk 2]
```diff
@@ -38,31 +25,36 @@ Give your repo a name and click on "Create repository":
...
+> If you are an existing GitHub user and have already set up a public/private key pair (or have done so by selecting `SSH` as the protocol when running `gh auth login`), you can also clone the repo as follows:
+> ```
+> $ git clone git@github.com/<username>/<reponame>.git
+> ```
 
 ## Configure Nix
 
+> **_Note for VM users_**
+>
+> If you are using the [VM image](https://vm.lf-lang.org/), you can skip this step. You will never have to invoke `nix` and can ignore any reminders about doing this.
```
Comment (lhstrh) [inline src/GettingStarted.md:56]: @perry0513 and @victoria-tuck, please confirm that this is indeed the case. If there is still any dependency missing in the VM, I will add it.

Comment (lhstrh) [reply src/GettingStarted.md:56]: I'm able to compile the examples after pulling in the submodules of `pico-sdk`. Will add those to the VM (hope it won't blow up too much), and publish it.

**Pull request merged**